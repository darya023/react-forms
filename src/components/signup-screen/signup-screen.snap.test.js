import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import SignupScreen from './signup-screen';

const mockStore = configureStore({});
const SendingStatus = {
  INITIAL: `INITIAL`,
  FETCHING: `FETCHING`,
  SUCCESS: `SUCCESS`,
  FAILURE: `FAILURE`,
};
const fakeStore = {
  user: [],
  sendingDataStatus: SendingStatus.FAILURE,
  authorizationStatus: null,
};
it(`SignupScreen should render correctly`, () => {
  const store = mockStore(fakeStore);
  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <SignupScreen />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
