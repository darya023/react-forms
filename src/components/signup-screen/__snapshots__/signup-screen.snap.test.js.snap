// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupScreen should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <a
        class="link"
        href="/"
      >
        На главную
      </a>
    </header>
    <main>
      <form
        action="#"
        class="form"
      >
        <h3
          class="form__title"
        >
          Регистрация
        </h3>
        <div
          class="form__fields"
        >
          <div
            class="form__row"
          >
            <label
              class="form__label"
              for="email"
            >
              Почта
            </label>
            <input
              class="form__input"
              id="email"
              name="email"
              placeholder="example@mail.ru"
              type="text"
              value=""
            />
          </div>
          <div
            class="form__row"
          >
            <label
              class="form__label"
              for="password"
            >
              Пароль
            </label>
            <input
              class="form__input"
              id="password"
              name="password"
              placeholder="Пароль"
              type="password"
              value=""
            />
          </div>
          <div
            class="form__row"
          >
            <label
              class="form__label"
              for="confirmed_password"
            >
              Повторите пароль
            </label>
            <input
              class="form__input"
              id="confirmed_password"
              name="confirmed_password"
              placeholder="Пароль"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="form__submit form__row"
        >
          <button
            class="button"
            type="submit"
          >
            Зарегистрироваться
          </button>
        </div>
        <div
          class="form__row"
        >
          <span>
            Есть аккаунт? 
          </span>
          <a
            class="link"
            href="/login"
          >
            Войти
          </a>
        </div>
      </form>
      Что-то пошло не так. Попробуйте позже.
    </main>
  </div>
</div>
`;
